src/main/java/io/castle/example/LoginServlet.java

```diff
@@ -1,38 +1,83 @@
 package io.castle.example;
 
 import io.castle.client.Castle;
-import io.castle.client.objects.Event;
-import io.castle.client.objects.UserInfoHeader;
+import io.castle.client.api.CastleApi;
+import io.castle.client.model.AuthenticateAction;
+import io.castle.client.model.Verdict;
+import io.castle.example.model.TestUser;
+import io.castle.example.model.UserAuthenticationBackend;
 
 import javax.servlet.ServletException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
 import java.io.IOException;
-import java.net.URI;
+import java.util.logging.Logger;
 
 @WebServlet("/login")
 public class LoginServlet extends HttpServlet {
 
+    private static final Logger LOGGER = Logger.getLogger(LoginServlet.class.getName());
+
     @Override
     protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
-        Castle.setAPISecret(Config.apiKey());
+        CastleApi castleApi = Castle.sdk().onRequest(req);
 
-        Event event = new Event();
-        event.setUserId("1234");
-        event.setName("$login.succeeded");
+        String overrideString = req.getParameter("override");
```

Comment:
I'm guessing you added this for test purposes. please remove the override functionality

Reply:
Done.